assume/markets/clearing_algorithms/simple.py

```diff
@@ -112,20 +112,17 @@ def clear(
                     split_demand_order["accepted_volume"] = diff
                     demand_order["accepted_volume"] = demand_order["volume"] - diff
                     rejected_orders.append(split_demand_order)
+
                 elif diff > 0:
                     # generation left over - split last generation bid
                     supply_order = to_commit[-1]
-                    split_supply_order = supply_order.copy()
-                    split_supply_order["volume"] = diff
                     supply_order["accepted_volume"] = supply_order["volume"] - diff
+
                     # changed supply_order is still part of to_commit and will be added
                     # only volume-diff can be sold for current price
                     gen_vol -= diff
 
-                    # add left over to supply_orders again
-                    supply_orders.insert(0, split_supply_order)
```

Comment:
We need to fix the market clearing in a different way..
As we are simply adding demand - then add supply until demand is matched -> then add again to the demand, we might have the split situation more than once per clearing.

The leftover will therefore not be traded at afterwards.

Reply:
This clearing is anyway not suitable for complex orders and other things. So I would day we remove this splitting to save bids for future and have a true simple clearing.

Reply:
I have reverted the clearing to the initial state as it works fine now anyway